packages/tests/components/SfRatingButton/SfRatingButton.cy.tsx

```diff
@@ -0,0 +1,159 @@
+/// <reference path="../../../../node_modules/@percy/cypress/types/index.d.ts" />
+import React from 'react';
+import { SfRatingButtonSize, noop } from '@storefront-ui/shared';
+import { mount, useComponent } from '../../utils/mount';
+
+const { vue: SfRatingButtonVue, react: SfRatingButtonReact } = useComponent('SfRatingButton');
+import SfRatingButtonBaseObject from './SfRatingButton.PageObject';
+
+describe('SfRatingButton', () => {
+  const page = () => new SfRatingButtonBaseObject('ratingbutton');
+
+  const initializeComponent = (
+    props: {
+      value?: number;
+      onChange?: (value: number) => void;
+      max?: number;
+      name?: string;
+      disabled?: boolean;
+      ariaLabel?: string;
+      size?: SfRatingButtonSize;
+      getLabelText?: (value: number) => string;
+      // children?: (state: SfRatingButtonRenderProps) => ReactNode;
+    } = {},
+  ) => {
+    const {
+      value = 0,
+      onChange = noop,
+      max = 5,
+      name = 'sf-rating-button',
+      disabled = false,
+      ariaLabel = 'Rating',
+      size = SfRatingButtonSize.base,
+      getLabelText = (value) => `${value} star`,
+    } = props;
+    return mount({
+      vue: {
+        component: SfRatingButtonVue,
+        props: {
+          modelValue: value,
+          'onUpdate:modelValue': onChange,
+          max,
+          name,
+          disabled,
+          ariaLabel,
+          size,
+          getLabelText,
+        },
+      },
+      react: (
+        <SfRatingButtonReact
+          value={value}
+          onChange={onChange}
+          max={max}
+          name={name}
+          disabled={disabled}
+          ariaLabel={ariaLabel}
+          size={size}
+          getLabelText={getLabelText}
+        ></SfRatingButtonReact>
+      ),
+    });
+  };
+
+  it('initial state', () => {
+    initializeComponent();
+
+    page().makeSnapshot();
+  });
+
+  describe('when value prop', () => {
+    it('should trigger onChange/emit modelValue when click on icon', () => {
+      const props = { value: 6, max: 9, onChange: cy.spy() };
+      initializeComponent(props);
+
+      page().clickEmptyIcon();
+      cy.then(() => {
+        expect(props.onChange).calledOnceWith();
```

Comment:
Called once with nothing? 😄 The value change should be tested here imo

Reply:
It should be `expect(spy).to.be.calledOnce`. Fixing...

> The value change should be tested here

What do you mean? To create a fake state, and try to click through and see the change?

Reply:
Actually we should do `expect(spy).calledOnceWith(expectedValue)`, to make sure `onChange` has been called with appropriate value.